docs/adr/0005-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,106 @@
+
+## Decision Drivers
+
+* We want to get changes merged and released to our consumer applications faster.
+* We want to simplify pushing new code to applications, especially urgent bug fixes.
+* We want to move towards the future monorepo world.
+* Mozilla is moving towards rapid releases for mobile code.
+* We are willing to accept the overhead of upgrading application-services more frequently in application repositories.
+
+## Considered Options
+
+* **(A) Keep our current release process**
+* **(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+    * Phase 1: update our branching model and release process (Jan 2023 - Feb 2023)
+        * Update our branches to match how firefox-android works: a `main` branch that is used for the current nightly builds and a `releases_v[N]` for each previous release.
+        * Automate a nightly release of our `main` branch.
+          * Create a commit that updates our version number to something like `[firefox-version].0-nightly-[YYYYMMDD]`, similar to what `prepare-release.py` does now.
+            * Make sure that the github CI can push this commit without needing PR approval.
+            * Note: we will no longer be using semver-style versions and also not following semver semantics.  We will be releasing breaking changes in a nightly cycle without bumping the minor version.
+          * Create a GitHub release from the branch with a tag based on the version number / date (`v[firefox-version].0-nightly-[YYYYMMDD]`). This will cause maven packages to be published, start the `rust-components-swift` CI, etc.
+        * Manually handle releases from the `release_v*` branches whenever new code is committed to those branches
+          * Update our version number to `[firefox-version].[N+1]`
+          * Create a GitHub release from the branch
+          * Create PRs to merge the changes forward to `main` and more recent release branches.
+        * Continue to maintain our CHANGELOG, but with new versioning.  In main, keep a rolling log of updates with the header `[firefox-version].0`
+        * This replaces our current release process -- no more cutting releases.
+        * To push changes to a consumer: create a PR in their repository that bumps the application-services version to an existing release.
+        * On the monthly Firefox merge date:
```

Comment:
We should automate this as much as possible, even if run manually. So maybe write something like "We will implement a script which will be manually run on the monthly Firefox merge date. This script will:" or similar?